swap/protocol_test.go

```diff
@@ -315,3 +320,150 @@ func TestTriggerDisconnectThreshold(t *testing.T) {
 		t.Fatalf("Expected still no cheque, but there are %d", lenCheques)
 	}
 }
+
+// TestSwapRPC tests some basic things over RPC
+// We want this so that we can check the API works
+func TestSwapRPC(t *testing.T) {
+
+	var (
+		p2pPort = 30100
+		ipcPath = ".swarm.ipc"
+		err     error
+	)
+
+	swap, testDir := newTestSwap(t)
+	defer os.RemoveAll(testDir)
+
+	// need to have a dummy contract or the call will fail at `GetParams` due to `NewAPI`
+	swap.contract, err = contract.InstanceAt(common.Address{}, swap.backend)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// start a service stack
+	stack := createAndStartSvcNode(swap, ipcPath, p2pPort, t)
+	defer stack.Stop()
+
+	// connect to the servicenode RPCs
+	rpcclient, err := rpc.Dial(filepath.Join(stack.DataDir(), ipcPath))
+	if err != nil {
+		t.Fatal("connect to servicenode IPC fail", "err", err)
+	}
+	defer os.RemoveAll(stack.DataDir())
+
+	// create dummy peers so that we can artificially set balances and query
+	dummyPeer1 := newDummyPeer()
+	dummyPeer2 := newDummyPeer()
+	id1 := dummyPeer1.ID()
+	id2 := dummyPeer2.ID()
+
+	// set some fake balances
+	fake1 := int64(234)
+	fake2 := int64(-100)
+
+	// query a first time, should be zero
+	var balance int64
+	err = rpcclient.Call(&balance, "swap_balance", id1)
+	if err != nil {
+		t.Fatal("servicenode RPC failed", "err", err)
+	}
+	log.Debug("servicenode balance", "balance", balance)
+
+	if balance != 0 {
+		t.Fatal("Expected balance to be 0 but it is not")
```

Comment:
Why not add the actual value of `balance` to the logs, for when it doesn't match the expected one?